<?xml version='1.0' encoding='UTF-8' standalone='no'?>
<section xmlns="http://docbook.org/ns/docbook" version="5.0" xmlns:xlink="http://www.w3.org/1999/xlink" xml:id="__i_o_exception_8hpp" xml:lang="en-US">
<title>src/runtime/IOException.hpp File Reference</title>
<indexterm><primary>src/runtime/IOException.hpp</primary></indexterm>
<programlisting linenumbering="unnumbered">#include &quot;RuntimeException.hpp&quot;<?linebreak?>#include &lt;jpl/utils/debug/ErrorUtils.hpp&gt;<?linebreak?></programlisting><simplesect>
    <title>Data Structures    </title>
        <itemizedlist>
            <listitem><para>class <link linkend="_classjpl_1_1__exception_1_1_i_o_exception">jpl::_exception::IOException</link></para>
</listitem>
        </itemizedlist>
</simplesect>
<simplesect>
    <title>Namespaces    </title>
        <itemizedlist>
            <listitem><para>namespace <link linkend="_namespacejpl">jpl</link></para>
</listitem>
            <listitem><para>namespace <link linkend="_namespacejpl_1_1__exception">jpl::_exception</link></para>
</listitem>
        </itemizedlist>
</simplesect>
<section>
<title>Detailed Description</title>

<para>An IOException is thrown when an error is occurred in runtime during an I/O operation either on a socket or a file</para>

<para>It is able to catch the last error occurred via _utils::_error::_GetLastError() if you instance via the default constructor; it even get the description of the error code via _utils::_error::_GetLastErrorAsString(unsigned int)</para>

<para><formalpara><title>Date</title>

<para>2023-08-01 </para>
</formalpara>
<formalpara><title>Copyright</title>

<para>Copyright (c) 2023 </para>
</formalpara>
<formalpara><title>Author</title>

<para>Gabriele-P03 </para>
</formalpara>
</para>
<para>
Definition in file <link linkend="__i_o_exception_8hpp_source">IOException.hpp</link>.</para>
</section>
</section>
